### PR TITLE
Funding content registry

### PR DESCRIPTION
**Pull request by ncjones**
The content registry exposes a rich interconnected domain model of all content relating to funding protocols. The content registry will be used to simplify the creation of various views on the related funding entities.

### CODE REVIEW

File: imsv-docs-astro/src/models/DeployedFundingProtocol.mjs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import path from 'node:path';
```
Comment (braden-holgate) [inline imsv-docs-astro/src/models/DeployedFundingProtocol.mjs:1]: Unused path module

File: imsv-docs-astro/src/models/ContentRegistry.mjs [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  getNetworkToken({ tokenName, networkName }) {
+    return this.getToken(tokenName).getInstance(networkName);
+  }
+
+  getNetwork(name) {
+    if (!this.#networksByName[name]) {
+      throw Error(`Unregistered network: ${name}`);
+    }
+    return this.#networksByName[name];
+  }
```
Comment (braden-holgate) [inline imsv-docs-astro/src/models/ContentRegistry.mjs:130]: Move this up so the pattern is consistent.
`registerObject`
`getObject`

File: imsv-docs-astro/src/models/ContentRegistry.mjs [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+  getFundingType(name) {
+    if (!this.#fundingTypesByName[name]) {
+      throw Error(`Unregistered funding type: ${name}`);
+    }
+    return this.#fundingTypesByName[name];
+  }
+
+  static async create() {
+    const registry = new ContentRegistry();
```
Comment (braden-holgate) [inline imsv-docs-astro/src/models/ContentRegistry.mjs:148]: Style preference. Spacing between the registry entry creation calls would make it more readable.

File: imsv-docs-astro/src/models/FundingProtocol.mjs [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import path from 'node:path';
+import { DeployedFundingProtocol } from './DeployedFundingProtocol.mjs';
```
Comment (braden-holgate) [inline imsv-docs-astro/src/models/FundingProtocol.mjs:2]: Where is this referenced? I see the type in a comment. Does this import need to be here?

Comment (ncjones) [reply imsv-docs-astro/src/models/FundingProtocol.mjs:2]: Yes, JSDoc needs the import.

**Review by braden-holgate — APPROVED** [review]
A+ . Very neat. Love the TDD

**Pull request merged**